include/cantera/thermo/PengRobinsonMFTP.h

```diff
@@ -0,0 +1,429 @@
+//! @file PengRobinsonMFTP.h
+
+// This file is part of Cantera. See License.txt in the top-level directory or
+// at https://cantera.org/license.txt for license and copyright information.
+
+#ifndef CT_PENGROBINSONMFTP_H
+#define CT_PENGROBINSONMFTP_H
+
+#include "MixtureFugacityTP.h"
+#include "cantera/base/Array.h"
+
+namespace Cantera
+{
+/**
+ * Implementation of a multi-species Peng-Robinson equation of state
+ *
+ * @ingroup thermoprops
+ */
+class PengRobinsonMFTP : public MixtureFugacityTP
```

Comment:
I know this parallels the naming convention of RedlichKwongMFTP, but I'd like to suggest dropping the incomprehensible acronym suffix and go with either PengRobinsonPhase or just PengRobinson if there's no potential chance of confusion with another type of object.

Reply:
Done.

Reply:
FYI, I'm in the middle of refactoring some of these common functions up to the `MixtureFugacityTP` level, but some (`setTemperature`, `compositionChanged`, to name two), already exist in the `MixtureFugacityTP` phase, and are not equal to the implementations in the lower-level classes.  Replacing them with the lower-level versions leads to build errors.

Reply:
If you want to push the code with a commit making this change and showing the error, I'd be happy to take a look at it.

Reply:
Okay, I just finished (https://github.com/Cantera/cantera/pull/641/commits/eb1dc63b016b52e93e7231fc283b6a315e0565d3) doing the "simpler" ones that did not lead to errors (some were build, some were test failures).

When I come back to it, I'll push one that errors, and also take a look at abstracting the solver.

Reply:
Okay, I've pushed this to my repo (https://github.com/decaluwe/cantera/commit/bbedc50156fc6914b66d38f73b316821f9972e7f), demonstrating the test failure in `PengRobinson_Test::getPressure()`.

When I move the `setTemperature` routine from `PengRobison` up to `MixtureFugacityTP`, the test (which sets temperature and density and then tests the calculated pressure vs. the EoS) fails.  This is the only test calling `setState_TR`; all tests involving calls to `setState_TP` succeed. 

I dug a little deeper, and the failure comes down to an error in the `alpha` variable (`a_coef` is correct, but `aAlpha` is not).  I tried a few things to fix it (changing the `Alpha_vec_Curr_` and `m_aAlpha_current` routines in `PengRobinson::updateAB`), but only seemed to make it worse.  Again, locating `setTemperature` within the `PengRobinson` class produces no test failures.

Any suggestions are greatly appreciated.

Reply:
Note also that I can remove `RedlichKwongMFTP::setTemperature` just fine, with no test failures (this class does not have a `getPressure` test, but it also does not have an `alpha` variable, so I feel confident that it would still pass, even if I implemented such a test).